### PR TITLE
Add library building of packages and included small helper-functions

### PR DESCRIPTION
**Pull request by ipa-foj**

### CODE REVIEW

File: ipa_building_navigation/CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,13 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(ipa_building_navigation)
 
+# switch build type
+set(ROS_BUILD_TYPE Debug)
```
Comment (ipa-rmb) [inline ipa_building_navigation/CMakeLists.txt]: Why in debug? Shouldn'd it build faster release code?

File: ipa_building_navigation/common/src/genetic_TSP.cpp [hunk 2]
```diff
@@ -213,7 +214,7 @@ std::vector<int> GeneticTSPSolver::solveGeneticTSP(const cv::Mat& path_length_Ma
 	if(path_length_Matrix.rows > 2) //check if graph has at least three members, if not the algorithm won't work properly
 	{
 		bool changed_path = false; //this variable checks if the path has been changed in the mutation process
-		int changeing_counter = 100; //this variable is a counter for how many times a path has been the same
+		int changeing_counter = const_generations_number_; //this variable is a counter for how many times a path has been the same
```
Comment (ipa-rmb) [inline ipa_building_navigation/common/src/genetic_TSP.cpp]: changing (Rechtschreibung :-D)

**Review by ipa-rmb — CHANGES_REQUESTED** [review]

**Review by ipa-rmb — CHANGES_REQUESTED** [review]
Travis is not building, please check. Furthermore, please harmonize the introduced spaces in the CMakeLists.txt files to tabs, I guess the remainder is all tabs.

**Comment by ipa-mtb** [comment]
Problem seems to be position independent code in ``libdlib`` in ``cob_extern``.
Might want to try out using this branch (https://github.com/ipa-mtb/cob_extern/tree/feature/fpic_dlib), if this would make travis happy

**Pull request closed without merging**